### PR TITLE
feat: support custom resource id

### PR DESCRIPTION
**Pull request by bzp2010**
### Description

Allow custom resource IDs for dump and sync.

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible

### CODE REVIEW

File: libs/backend-apisix/e2e/misc.e2e-spec.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+          'custom-route',
+          'custom-service',
+        ),
+      ]));
+
+    it('Dump', async () => {
+      const result = (await dumpConfiguration(backend)) as ADCSDK.Configuration;
+      expect(result.services).toHaveLength(1);
+      expect(result.services[0]).toMatchObject(service);
+      expect(result.services[0]).toMatchObject(service);
```
Comment (juzhiyuan) [inline libs/backend-apisix/e2e/misc.e2e-spec.ts:64]: double check?

File: libs/backend-apisix/e2e/support/utils.ts [hunk 2]
```diff
@@ -91,6 +91,16 @@ export const deleteEvent = (
     : undefined,
 });
 
+export const overrideEventResourceId = (
```
Comment (juzhiyuan) [inline libs/backend-apisix/e2e/support/utils.ts:94]: https://github.com/api7/adc/pull/203/files#diff-af2b7c68fc5660ece2df03f69e1ff4bb37ac1e2b28f5a11549546e1a10e773b0R105

should we only keep one function?

File: libs/backend-apisix/e2e/misc.e2e-spec.ts (hunk 1 above)
Comment (bzp2010) [reply libs/backend-apisix/e2e/misc.e2e-spec.ts]: fixed

File: libs/backend-apisix/e2e/support/utils.ts (hunk 2 above)
Comment (bzp2010) [reply libs/backend-apisix/e2e/support/utils.ts:94]: They are in different packages and have difficulties in referencing each other, which requires another separate package to place the public functions; and the backends should not depend on each other, so they are kept first.

**Review by juzhiyuan — APPROVED** [review]

**Pull request merged**